molten-cache/src/main/java/com/hotels/molten/cache/ReactiveCache.java

```diff
@@ -43,4 +45,46 @@
      * @return a completion once the operation has finished
      */
     Mono<Void> put(K key, V value);
+
+    /**
+     * Return the value stored to {@code key} if any,
+     * otherwise subscribe to the given {@code Mono}
+     * and store its return value in the cache.
+     *
+     * <pre>
+     * return Mono.fromCallable(() -&gt; expensiveServiceCall(key))
```

Comment:
I think the example should promote a reactive mono as wrapped one.

Reply:
Not sure if I understand what you mean.

Reply:
I mean `expensiveServiceCall` is a blocking operation wrapped with `fromCallable` to make it sort of reactive.